src/modules/twap/hooks/useAreWarningsAccepted.ts

```diff
@@ -0,0 +1,18 @@
+import { useAtomValue } from 'jotai'
+
+import { useIsFallbackHandlerRequired } from './useFallbackHandlerVerification'
+import { useTwapWarningsContext } from './useTwapWarningsContext'
+
+import { twapOrdersSettingsAtom } from '../state/twapOrdersSettingsAtom'
+
+export function useAreWarningsAccepted(): boolean {
+  const { isPriceImpactAccepted, isFallbackHandlerSetupAccepted } = useAtomValue(twapOrdersSettingsAtom)
+  const { showPriceImpactWarning } = useTwapWarningsContext()
+  const isFallbackHandlerRequired = useIsFallbackHandlerRequired()
+
+  if (showPriceImpactWarning) return isPriceImpactAccepted
```

Comment:
not sure this is what you wanted to do

Didn;t you mean sth like `if (showPriceImpactWarning && !isPriceImpactAccepted) return false`

Reply:
In this hook, I like to take into account `isPriceImpactAccepted` value only when `showPriceImpactWarning` is true.
So, I can translate it as:
```
When price impact is shown, then take the "Price impact accepted" checkbox value into account
```

Reply:
But can it be `showPriceImpactWarning` and `isFallbackHandlerRequired` 

in my proposal above it says, "if we show the price impact and is not accepted, return false, otherwise continue with the following logic" this way we consider other checks. If all check pass then we return true

Maybe I'm reading it wrong?